storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +103,52 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	dotGitobjcts := d.fs.Join(d.path, "objects")
+	files, err := d.fs.ReadDir(dotGitobjcts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var objDirs []string
+	reDir, _ := regexp.Compile("[a-z0-9]{2}")
+	for _, f := range files {
+		if f.IsDir() && reDir.MatchString(f.Name()) {
+			objDirs = append(objDirs, f.Name())
+		}
+	}
+
+	var objects []core.Hash
+	reObj, _ := regexp.Compile("[a-z0-9]{38}")
+	for _, dir := range objDirs {
```

Comment:
why are we looping twice? one to fill an slice of dirs, and then again to fill the slice of objects.

We don't really need the slice of objDirs, just visit the dirs if they match the regexp and fill the objects slice inplace.